internal/state/graph/reference_grant.go

```diff
@@ -0,0 +1,64 @@
+package graph
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func refGrantAllowsGatewayToSecret(
```

Comment:
would it make sense to have more generalized function to support more types of references in the future?

for example,

```
allowCrossNamespaceReference(from, to client.Object) bool
```

Reply:
Yes, but I was waiting to generalize until I started the next ReferenceGrant issue.